cranelift/codegen/src/isa/x64/inst.isle

```diff
@@ -3483,9 +3512,7 @@
 
 ;; Result = (a_hi <> b_hi) ||
 ;;          (a_hi == b_hi && a_lo <> b_lo)
-(rule (emit_cmp cc a @ (value_type $I128) b)
-      (if (intcc_neq cc (IntCC.Equal)))
-      (if (intcc_neq cc (IntCC.NotEqual)))
+(rule 2 (emit_cmp (and (intcc_is_ordered $true) cc) a @ (value_type $I128) b)
```

Comment:
I think that would be fine, however the rule that I've been trying to follow is to use priorities as a fall back for when there's not a clear way to split the rules by a concrete pattern. Maybe there's a different name we could use here that would make it seem less objectionable?

Reply:
That's not what I was suggesting for the use of or-patterns: I think we would not have the `and` anymore for the `Equal` and NotEqual` cases, and instead would list the remaining cases where we have `(intcc_is_ordered $true)` now. That feels pretty natural to me, and is likely how you'd write it in rust if you wanted to enumerate all the cases, right?

Reply:
At any rate, I've removed `intcc_is_ordered`, and put explicit priorities in place instead. We can revisit this if we decide to implement or-patterns in the future.

Reply:
Oh, sorry, I wasn't clear: I know you weren't suggesting expanding `intcc_is_ordered` to an or-pattern. I just meant that's how I had to reason about what the and-patterns were doing, and it was a confusing thing to reason about.

I agree, listing the remaining cases with an or-pattern instead of priorities feels pretty natural to me, too. Only it's a bit of a shame that we can't check for match completeness, in case somebody accidentally misses a condition code.